### ISSUE TITLE
Cannot upload to /

### ISSUE THREAD

**Issue by cyqsimon** [thecoshman/http#125]
So, 3c781ab introduced another issue, namely that you now cannot upload files to `/`.

This is what happens when you upload a file with no extension (`some_file`):

![no_ext](https://user-images.githubusercontent.com/28627918/106602165-963d5100-6597-11eb-970b-3104f1bae724.png)

And a bit scarier when it's a file with extension (`some.file`):

![ext](https://user-images.githubusercontent.com/28627918/106602209-a5bc9a00-6597-11eb-9577-83e0de612097.png)

Yikes. Fortunately it's caught by CORS, else my files are going somewhere.

<hr>

The root cause is that it's not safe to do this:
```js
let url = document.location.pathname; // p.s. the var really shouldn't be named "url", rather "path".
// See https://nodejs.org/api/url.html#url_url_strings_and_url_objects

upload_file(url + "/" + encodeURIComponent(file.name), file);
```
Path might very well end with `/`, and it's not safe to assume that `//` is going to behave like `/`.

The simplest and best example is of course `/` (the base path). In this case, you get `"//file_name"`, which might be interpreted as `<proto>://file_name/`, where `host == "file_name"` and `path == "/"`, leading to the aforementioned errors.

There's also the case of `//////`, `/foo/` or `path == "/bar///////"`, which should all be considered.

Which is why in #124 I wrote this code like this:
```js
let page_path = document.location.pathname.replace(/\/+$/, "") + "/"; // always leaves one and only one trailing "/"
// can't do .replace(/\/+$/, "/") because there might be no match.
upload_file(page_path + encodeURIComponent(file.name), file);
```

Should be a simple fix.

**Comment by cyqsimon** [comment]
And btw the reason I personally prefer using `document.location.pathname` rather than `document.URL` is that it's convention to omit the origin in HTML/JS unless writing cross-site code.

**Comment by nabijaczleweli** [comment]
Try with current `develop` (at least 589a99264bc171ad3300305ab3622aea45bc7c05)?

**Comment by cyqsimon** [comment]
Yep, working as intended.

However, instead of [this](https://github.com/thecoshman/http/commit/589a99264bc171ad3300305ab3622aea45bc7c05#diff-1eeb728d7f22db1c6a31dd465b3d8b8e7fdd618c3ea35388b3abc795b02cc865R65-R69):
```js
upload_file(base_url + entry.fullPath.split("/").filter(function(seg) { return seg; }).map(encodeURIComponent).join("/"), f);
```
Correct me if I'm wroing, but I think you meant this?
```js
upload_file(
  (base_url + entry.fullPath.split("/").map(encodeURIComponent).join("/"))
    .replace(/\/\/+/, "/"), // matches two or more consecutive "/" and replaces with a single "/"
  f
);
```
I think I understand that you intend to get rid of repeating `/`s by dropping any empty strings. The reason I'm asking is `entry` is a [`FileSystemEntry`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemEntry), which will not have consecutive `/`s. (I certainly hope not, otherwise it would be a huge browser bug)

Also I always encourage people to use more regex. It's so powerful 😄

**Comment by nabijaczleweli** [comment]
No, I meant exactly to remove empty segments. On my Firefox the path is `/hehe.hoho`, which split to `["", "hehe.hoho"]`, the predicate yields just `["hehe.hoho"]`, which, when joined, doesn't have a leading slash. Ideally I'd just remove the first slash regardless, but that's not one-liner friendly. Consecutive slashes are fine, so long as they aren't at the start, but that gets them as well.

Good, this means this is ready to ship, thanks for testing.

**Comment by nabijaczleweli** [comment]
Released in [`v1.11.1`](https://github.com/thecoshman/http/releases/tag/v1.11.1).
